tests/src/test/scala/com/nvidia/spark/rapids/RegularExpressionTranspilerSuite.scala

```diff
@@ -436,6 +436,26 @@ class RegularExpressionTranspilerSuite extends FunSuite with Arm {
     doAstFuzzTest(Some(REGEXP_LIMITED_CHARS_REPLACE), RegexReplaceMode)
   }
 
+  test("string split - optimized") {
+    val patterns = Set("\\.", "\\$", "\\[", "\\(", "\\}", "\\+", "\\\\", "c\\|d")
```

Comment:
It would be good to add some patterns without escaped characters as well, such as `","`

Reply:
Added a few more tests. Note though that the original check for regular expression will probably supersede the check here. 

A thought might be to remove the isRegExpString call in the first place, and just treat non-regexp strings with the same code path. Would that make sense?

Reply:
I pushed a change to consolidate this logic a bit to remove redundancy.